File: Services/Student/Student.Application/Handlers/GetCourseQueryHandler.cs
using AutoMapper;
using MediatR;
using Microsoft.Extensions.Logging;
using Student.Application.Queries;
using Student.Application.Dtos;
using Student.Core.Repositories;

namespace Student.Application.Handlers;

public class GetCourseQueryHandler : IRequestHandler<GetCourseQuery, ResponseDto<CourseDto>>
{
    private readonly IMapper _mapper;
    private readonly ILogger<GetCourseQueryHandler> _logger;
    private readonly ICourseRepository _courseRepository;
    public GetCourseQueryHandler(ICourseRepository courseRepository, IMapper mapper, ILogger<GetCourseQueryHandler> logger)
    {
        _courseRepository = courseRepository;
        _mapper = mapper;
        _logger = logger;
    }

    public async Task<ResponseDto<CourseDto>> Handle(GetCourseQuery request, CancellationToken cancellationToken)
    {
        var course = await _courseRepository.Get(request.Id);
        var result = _mapper.Map<CourseDto>(course);
        return new ResponseDto<CourseDto>
        {
            Success = result is not null,
            Message = result is not null ? "" : "No data found",
            Data = result
        };
    }
}

File: Services/Teacher/Teacher.Application/Handler/GetCourseQueryHandler.vb
Imports System.Threading
Imports MediatR
Imports Teacher.Core

Public Class GetCourseQueryHandler
    Implements IRequestHandler(Of GetCourseQuery, ResponseDto(Of CourseDto))
    Private ReadOnly _courseRepository As ICourseRepository

    Public Sub New(courseRepository As ICourseRepository)
        _courseRepository = courseRepository
    End Sub

    Public Async Function Handle(request As GetCourseQuery, cancellationToken As CancellationToken) As Task(Of ResponseDto(Of CourseDto)) Implements IRequestHandler(Of GetCourseQuery,ResponseDto(Of CourseDto)).Handle
        Dim course = Await _courseRepository.GetById(request.Id)
        Return New ResponseDto(Of CourseDto) With {
            .Success = course IsNot Nothing,
            .Message = If(course IsNot Nothing, "", "No data found"),
            .Data = New CourseDto With{
                .Id = course.Id,
                .Name = course.Name,
                .Description = course.Description
                }
            }
    End Function
End Class